services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -577,47 +648,60 @@ def compute_descriptive_statistics_response(
         local_parquet_glob_path, columns=[pl.scan_parquet(local_parquet_glob_path).columns[0]]
     ).shape[0]
 
-    class_label_features = {
-        feature_name: feature
-        for feature_name, feature in features.items()
-        if isinstance(feature, dict) and feature.get("_type") == "ClassLabel"
-    }
-    numerical_features = {
-        feature_name: feature
-        for feature_name, feature in features.items()
-        if isinstance(feature, dict) and feature.get("_type") == "Value" and feature.get("dtype") in NUMERICAL_DTYPES
-    }
-    string_features = {
-        feature_name: feature
-        for feature_name, feature in features.items()
-        if isinstance(feature, dict) and feature.get("_type") == "Value" and feature.get("dtype") in STRING_DTYPES
-    }
-    bool_features = {
-        feature_name: feature
-        for feature_name, feature in features.items()
-        if isinstance(feature, dict) and feature.get("_type") == "Value" and feature.get("dtype") == "bool"
-    }
-    if not class_label_features and not numerical_features and not string_features:
+    def _column_from_feature(
+        dataset_feature_name: str, dataset_feature: Union[dict[str, Any], list[Any]]
+    ) -> Optional[SupportedColumns]:
+        if isinstance(dataset_feature, list):
+            return ListColumn(feature_name=dataset_feature_name, n_samples=num_examples, n_bins=histogram_num_bins)
+
+        if isinstance(dataset_feature, dict):
+            if dataset_feature.get("_type") == "Sequence":
+                return ListColumn(feature_name=dataset_feature_name, n_samples=num_examples, n_bins=histogram_num_bins)
+
+            if dataset_feature.get("_type") == "ClassLabel":
+                return ClassLabelColumn(
+                    feature_name=dataset_feature_name, n_samples=num_examples, feature_dict=dataset_feature
+                )
+
+            if dataset_feature.get("_type") == "Value":
+                if dataset_feature.get("dtype") in INTEGER_DTYPES:
+                    return IntColumn(
+                        feature_name=dataset_feature_name, n_samples=num_examples, n_bins=histogram_num_bins
+                    )
+
+                if dataset_feature.get("dtype") in FLOAT_DTYPES:
+                    return FloatColumn(
+                        feature_name=dataset_feature_name, n_samples=num_examples, n_bins=histogram_num_bins
+                    )
+
+                if dataset_feature.get("dtype") in STRING_DTYPES:
+                    return StringColumn(
+                        feature_name=dataset_feature_name, n_samples=num_examples, n_bins=histogram_num_bins
+                    )
+
+                if dataset_feature.get("dtype") == "bool":
+                    return BoolColumn(feature_name=dataset_feature_name, n_samples=num_examples)
+        return None
+
+    all_stats = []
+    for feature_name, feature in features.items():
```

Comment:
do you think we could run them in parallel? would it help?

Reply:
Maybe but I wouldn't do this to avoid potential fights between Polars' threads and python processes because Polars is multithreaded when possible (I still have nightmares with multithreaded duckdb in a subprocess :D)

I assume it would be more efficient to rewrite code somehow to apply processing funcs to all columns (or maybe all columns of the same type) at once using native Polars code but I think this is a pretty massive change... 

for reference: https://docs.pola.rs/user-guide/misc/multiprocessing/

Reply:
> or maybe all columns of the same type

i think this is actually quite possible but maybe in another pr if we see that jobs stuck?